### PR TITLE
Make latejoining not hardcoded to presets

### PR DESCRIPTION
**Pull request by DTanxxx**
Made `DisallowLateJoin` a get&set property in GamePreset with a default value of false.

### CODE REVIEW

File: Content.Server/GameTicking/GamePreset.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public abstract class GamePreset
         public abstract bool Start(IReadOnlyList<IPlayerSession> readyPlayers, bool force = false);
         public virtual string ModeTitle => "Sandbox";
         public virtual string Description => "Secret!";
-        public virtual bool DisallowLateJoin => false;
+        public virtual bool DisallowLateJoin { get; set; } = false;
```
Comment (Zumorica) [inline Content.Server/GameTicking/GamePreset.cs:15]: Revert this change

File: Content.Server/GameTicking/GamePresets/PresetSuspicion.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ public class PresetSuspicion : GamePreset
         public int TraitorStartingBalance { get; set; }
 
 
-        public override bool DisallowLateJoin => true;
+        public override bool DisallowLateJoin { get; set; } = true;
```
Comment (Zumorica) [inline Content.Server/GameTicking/GamePresets/PresetSuspicion.cs:43]: Also revert this

**Review by Zumorica — CHANGES_REQUESTED** [review]
Changing this to a property in the presets will do nothing.
You need to add an admin command that changes `DisallowLateJoin` in the GameTicker and afterwards calls `UpdateLateJoinStatus()`

**Comment by DTanxxx** [comment]
Hi Zumorica,
By adding an 'admin command', does that mean adding an IClientCommand to GameTickerCommands.cs?

**Comment by Zumorica** [comment]
> Hi Zumorica,
> By adding an 'admin command', does that mean adding an IClientCommand to GameTickerCommands.cs?

Pretty much!

File: Content.Server/GameTicking/GameTickerCommands.cs [hunk 3]
```diff
@@ -244,6 +244,19 @@ public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
         }
     }
 
+    class ToggleDisallowLateJoinCommand: IClientCommand
+    {
+        public string Command => "toggledisallowlatejoin";
+        public string Description => "";
```
Comment (Zumorica) [inline Content.Server/GameTicking/GameTickerCommands.cs:250]: This should have a description saying what this command does

File: Content.Server/GameTicking/GameTickerCommands.cs [hunk 4]
```diff
@@ -244,6 +244,19 @@ public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
...
+        public string Help => "";
```
Comment (Zumorica) [inline Content.Server/GameTicking/GameTickerCommands.cs:251]: This should tell you the arguments you need to pass to the command

File: Content.Server/GameTicking/GameTickerCommands.cs [hunk 5]
```diff
@@ -244,6 +244,19 @@ public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
...
+
+        public void Execute(IConsoleShell shell, IPlayerSession player, string[] args)
+        {
+            var ticker = IoCManager.Resolve<IGameTicker>();
+            ticker.ToggleDisallowLateJoin(bool.Parse(args[0]));
```
Comment (Zumorica) [inline Content.Server/GameTicking/GameTickerCommands.cs:256]: You should only execute the command if you actually have the arguments (check the length of the args array and return if it's not enough)

**Review by Zumorica — APPROVED** [review]

**Pull request merged**